### ISSUE TITLE
Assert in xcb_io.c / seg fault when rapidly switching screens

### ISSUE THREAD

**Issue by james-fowler** [synergy/synergy#5157]
When moving the mouse rapidly across screens, my synergy client is crashing  

    ....
    [2015-12-15T13:30:35] DEBUG: open clipboard 0
    [2015-12-15T13:30:35] INFO: entering screen
    synergyc: ../../src/xcb_io.c:254: poll_for_event: Assertion `dpy->xcb->event_owner ==  XlibOwnsEventQueue && !dpy->xcb->event_waiter' failed.
    Aborted

or occasionally 

    ....
    [2015-12-15T13:48:18] DEBUG: open clipboard 1
    [2015-12-15T13:48:18] DEBUG: close clipboard 1
    [2015-12-15T13:48:18] INFO: entering screen
    Segmentation fault

This is more or less repeatable for me, under the following setup:

* both client and server are run with ***-f -d DEBUG***
* big chunk of text (around a dozen lines from a terminal) is copied to clipboard on the client
* server is left of client
* spin mouse (trackball with pointer speed maxed in settings) rapidly left to right, traversing across both systems and wrapping around several times a second
* keep trying, usually will happen within thirty seconds (sometimes happens much sooner, originally caught this by accident while tweaking pointer speed)
* both client and server are Ubuntu 15.10 
* client is fresh unmodified build of the current synergy [master](https://github.com/synergy/synergy/commit/7a207b4a54bf0b9c0199cfc04f9d44497a0ee372)
* server is built from https://github.com/james-fowler/synergy/tree/FixClipboard

Smells like something timing related, possibly with multi-step clipboard stuff getting started on entry into the client but not having time to complete before leaving (and potentially wrapping back in)? 

It's plausible that the increased debug output is introducing delay which contributes to the issue, however I was able to duplicate the seg fault without it (***synergyc -f server_name***).  

I've reproduced this dozens of times.  I have only seen it crash the client (not the server), and I'm fairly certain it only happens when the client has set the clipboard.  Spent a few minutes trying to break it with clipboard copied from the server side and it wouldn't break.

I would expect this to occur exceedingly rarely, if at all, in normal use - there's not a lot of practical need for spinning your cursor through all your screens several times a second - but it still would be nice to understand why this is happening, and ideally get it fixed. 

-- James

**Comment by XinyuHou** [comment]
Hi James,

Please try 1.7.6 branch

--
Xinyu Hou (Jerry)
Senior Software Engineer

**Comment by james-fowler** [comment]
So far I have not been able to reproduce the crash with a fresh build of the client from v1.7.6.   I did encounter something strange where my mouse buttons just quit working, even after shutting down synergy - actually had to reboot to get things working again, but this wasn't necessarily a synergy issue.  I also noticed that some of the issues I've been having with cut & paste (copy text on client - paste into Eclipse on server failed) appear to be resolved, so that's a bonus...

So is the v1.7.6 branch eventually going to be merged back to master?
